homeassistant/components/panasonic_viera/media_player.py

```diff
@@ -117,6 +177,14 @@ def __init__(self, mac, name, remote, host, broadcast, app_power, uuid=None):
         self._volume = 0
         self._app_power = app_power
 
+        hass.services.async_register(
```

Comment:
We don't want side effects in init method. Please move the service registration out of init.

Also, we should limit this pull request to one thing. A new service can be added in a different PR.